LambdaEngine/Include/Application/API/Events/EventQueue.h

```diff
@@ -4,259 +4,91 @@
 
 #include "Containers/TUniquePtr.h"
 
+#include "Memory/API/Malloc.h"
+#include "Memory/API/StackAllocator.h"
+
 namespace LambdaEngine
 {
 	/*
 	* EventContainer
 	*/
-	struct EventContainer
-	{
-	public:
-		using Iterator = Event*;
-		using ConstIterator = const Event*;
-
-		virtual ~EventContainer() = default;
-
-		inline EventContainer(EventType eventType) noexcept
-			: ContainerEventType(eventType)
-		{
-		}
 
-		virtual void Push(const Event& event) = 0;
-		virtual void Clear() = 0;
-		
-		virtual uint32 Size() const = 0;
-
-		virtual EventContainer* Copy(void* pMemory) = 0;
-		virtual EventContainer* Move(void* pMemory) = 0;
-		
-		virtual Event& GetAt(uint32 index) = 0;
-		virtual const Event& GetAt(uint32 index) const = 0;
-
-	public:
-		EventType ContainerEventType;
-	};
-
-	/*
-	* TEventContainer
-	*/
-	template<typename TEvent>
-	struct TEventContainer : public EventContainer
+	class EventContainer
 	{
 	public:
-		inline TEventContainer(uint32 size = 1) noexcept
-			: EventContainer(TEvent::GetStaticType())
-			, Events()
-		{
-			Events.Reserve(size);
-		}
-
-		inline TEventContainer(const TEventContainer& other) noexcept
-			: EventContainer(TEvent::GetStaticType())
-			, Events(other.Events)
+		inline EventContainer()
+			: m_Allocator()
+			, m_Events()
 		{
 		}
 
-		inline TEventContainer(TEventContainer&& other) noexcept
-			: EventContainer(TEvent::GetStaticType())
-			, Events(std::move(other.Events))
-		{
-		}
-
-		virtual void Push(const Event& event) override final
-		{
-			VALIDATE(ContainerEventType == event.GetType());
-			Events.EmplaceBack(EventCast<TEvent>(event));
-		}
+		~EventContainer() = default;
 
-		virtual void Clear() override final
-		{
-			Events.Clear();
-		}
-
-		virtual uint32 Size() const override final
-		{
-			return static_cast<uint32>(Events.GetSize());
-		}
-
-		virtual EventContainer* Copy(void* pMemory)
+		template<typename TEvent>
+		FORCEINLINE void Push(const TEvent& event)
 		{
-			new(pMemory) TEventContainer(*this);
-			return reinterpret_cast<EventContainer*>(pMemory);
+			void* pMemory = m_Allocator.Push(sizeof(TEvent));
+			new(pMemory) TEvent(event);
```

Comment:
Does this copy `event` into `pMemory`?

Reply:
Yes and no, this is a placement new, i.e construct a new object without allocating new memory, but we are using the copy constructor so yes, there will be a copy.

Reply:
Short answer yes